python/cudf/cudf/_lib/scalar.pyx

```diff
@@ -359,3 +414,11 @@ def as_scalar(val, dtype=None):
             return Scalar(val.value, dtype)
     else:
         return Scalar(value=val, dtype=dtype)
+
+def is_null_host_scalar(slr):
+    if slr is None or slr is cudf.NA:
+        return True
+    elif isinstance(slr, (np.datetime64, np.timedelta64)) and np.isnat(slr):
```

Comment:
Should we handle `pd.Timestamp` types in here as well?

Reply:
Turns out this was a source of some bugs and needed to be adjusted. A `NaT` scalar isn't necessarily invalid and behaves differently in `fillna` and I think a few other places.